### PR TITLE
Solution for hm22

### PR DESCRIPTION
**Pull request by MaksymMatuliak**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,18 +6,54 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        if (matrix.length == 1) {
+            return matrix[0];
+        }
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: try to solve without this `if`

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,18 +6,54 @@
...
+        int[] diagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[i] += matrix[i][i];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
            diagonal[i] = matrix[i][i];
```

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,18 +6,54 @@
...
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        if (matrix.length == 1) {
+            return matrix[0];
+        }
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: try to solve without this `if`

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -6,18 +6,54 @@
...
+        return diagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        if (matrix.length == 1) {
+            return matrix[0][0];
+        }
+        int min = Integer.MAX_VALUE;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: don't use Integer.MAX_Value

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 5]
```diff
@@ -6,18 +6,54 @@
...
+            diagonal[i] += matrix[i][matrix.length - 1 - i];
+        }
+        return diagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        if (matrix.length == 1) {
+            return matrix[0][0];
+        }
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: try to solve without this `if`

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 6]
```diff
@@ -6,18 +6,54 @@
...
+            }
+        }
+        return min;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        if (matrix.length == 1) {
+            return matrix[0][0];
+        }
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: try to solve without this `if`

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 7]
```diff
@@ -6,18 +6,54 @@
...
+        int max = Integer.MIN_VALUE;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: don't use Integer.MIN_Value

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 3 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 4 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 6 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 7 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 5 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**